inc/templates/namespace.php

```diff
@@ -130,56 +130,62 @@ function get_template_path( WP_Query $query ): string {
 	return apply_filters( 'wp_irving_template_include', $template, $query );
 }
 
+
+
 /**
  * Prepares data for an Irving REST response from a template.
  *
- * @param string $template Full path to template.
- * @param string $type     Optional. The type of data being loaded from a template.
- *                         Defaults to 'page'.
+ * @param string $template_path Full path to template.
  * @return array An associative array prepared for an Irving REST Response.
  */
-function prepare_data_from_template( string $template, string $type = 'page' ): array {
-	$components = [];
+function prepare_data_from_template( string $template_path ): array {
+	$data = [];
 
+	$type = pathinfo( $template_path, PATHINFO_EXTENSION );
+
+	// Load template data.
 	ob_start();
-	include $template;
+	include $template_path;
 	$contents = ob_get_clean();
 
-	// This is a .json template.
-	if ( false !== strpos( $template, '.json' ) ) {
-
-		// Attempt to json decode it.
-		$components = json_decode( $contents, true );
-
-		// Validate success.
-		if ( ! json_last_error() ) {
-			return $components;
-		}
-
-		wp_die(
-			sprintf(
-				// Translators: %1$s: Error message, %2$s Template path.
-				esc_html__( 'Error: %1$s found in %2$s.', 'wp-irving' ),
-				esc_html( json_last_error_msg() ),
-				esc_html( $template )
-			)
-		);
-	}
-
-	if ( has_blocks( $contents ) ) {
```

Comment:
I want to revisit block parsing later when we look at Gutenberg integrations. Prioritizing JSON templates during this run.